### PR TITLE
Summarize build execution times in CI

### PR DESCRIPTION
**Pull request by victorlin**
### Description of proposed changes

Add a script that extracts useful information from BuildKit output, use it on all 3 stages of the Docker build, and put it in the GitHub job summary.

### Related issue(s)

N/A, motivated by myself getting frustrated at how hard it is to inspect the logs for layer build times.

### Testing

- [x] Checks pass
- [x] Job summary for CI run looks good ([ref](https://github.com/nextstrain/docker-base/actions/runs/5382950654#summary-14569646765))
- [x] Job summary works for incomplete builds ([ref](https://github.com/nextstrain/docker-base/actions/runs/5383025237#summary-14570773300))

### CODE REVIEW

File: devel/summarize-buildkit-output [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    if incomplete:
+        warn("WARNING: This log is from an incomplete build.")
+
+    return id_map
+
+
+def print_summary(id_map):
+    """Print a summary of execution times per layer.
+
+    Bar charts inspired by https://alexwlchan.net/2018/ascii-bar-charts/
```
Comment (tsibley) [inline devel/summarize-buildkit-output:78]: _non-blocking_

I'm amused that blog post dubs these "ASCII" when in fact it's using Unicode block drawing chars. :upside_down_face:

**Review by tsibley — APPROVED** [review]
LGTM on a quick skim. Nice output!

It's too bad that BuildKit doesn't expose this information as JSON via a flag.

File: devel/summarize-buildkit-output [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    """
+    start_pattern = re.compile(r'^(#\d+) \[(.*?)\] (.*)$')
+    cached_pattern = re.compile(r'^(#\d+) CACHED$')
+    done_pattern = re.compile(r'^(#\d+) DONE (\d+.\d+s)$')
+
+    id_map = {}
+    for line in f:
+        start_match = start_pattern.match(line)
+        if start_match:
+            layer_id = start_match.group(1)
```
[the remark this answers is not in the document]
Comment (victorlin) [reply devel/summarize-buildkit-output]: Done in [force-push](https://github.com/nextstrain/docker-base/compare/c56a651aad0f37c7bc76431745fd787c8599d131..60393f0a21f9bfbbfbc5fb994568fb5cff76b2b9).

File: devel/summarize-buildkit-output [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+    }
+    """
+    start_pattern = re.compile(r'^(#\d+) \[(.*?)\] (.*)$')
+    cached_pattern = re.compile(r'^(#\d+) CACHED$')
+    done_pattern = re.compile(r'^(#\d+) DONE (\d+.\d+s)$')
+
+    id_map = {}
+    for line in f:
+        start_match = start_pattern.match(line)
+        if start_match:
```
[the remark this answers is not in the document]
Comment (victorlin) [reply devel/summarize-buildkit-output]: Done: 6eb6775e68a0feee7b3a0d74b290cdfd05f2749e

**Pull request merged**